itou/users/models.py

```diff
@@ -368,6 +359,12 @@ def save(self, *args, **kwargs):
         self.validate_constraints()
         self.validate_identity_provider()
 
+        # Capture Django update_last_login signal
+        if self.first_login is None and self.last_login is not None:
+            self.first_login = self.last_login
+            if "update_fields" in kwargs and "last_login" in kwargs["update_fields"]:
+                kwargs["update_fields"].append("first_login")
```

Comment:
Je me dit que c'est peut-être trop "naif" mais j'ai l'impression que ce qu'on veux c'est ça, non ?
```suggestion
        if self.first_login is None:
            self.first_login = self.last_login
            if "update_fields" in kwargs:
                kwargs["update_fields"].append("first_login")
```
- Si pas de dernière connexion, alors normal de ne pas avoir de première non plus.
- Si dernière connexion et pas de donnée alors on remplis
- Sinon la donnée est déjà présente donc on ignore.

Reply:
oui, c'est naif exprès.
Sachant que last_login n'est remplis que par le signal de django (et certaines de nos factories :))

Reply:
Les deux versions me vont. Je suis un peu chagrin de ne pas comprendre le soucis avec le signal mais je n'ai pas réussi à reproduire le soucis :-/